stdlib/ballerina-database/src/main/java/org/ballerinalang/database/sql/SQLDataIterator.java

```diff
@@ -255,14 +437,54 @@ public BStruct generateNext() {
                     }
                 }
             }
-        }  catch (Throwable e) {
+        } catch (Throwable e) {
             throw new BallerinaException(
                     "error in retrieving next value for column: " + columnName + ": at index:" + index + ":" + e
                             .getMessage());
         }
         return bStruct;
     }
 
+    private void handleUnionTypeStructField(BStruct bStruct, BStructType.StructField[] structFields, int index,
+            int refRegIndex, BRefType value, int refType) {
+        if (isNillableType(structFields, index - 1, refType)) {
+            bStruct.setRefField(refRegIndex, value);
+        } else {
+            throw new BallerinaException("Union type contains more than 2 members or the members do not result in "
+                    + "an assignable nillable type");
```

Comment:
Shall we rephrase this to  something like "Corresponding Union type in the record is not an assignable nillable type" as "Union type contains more than 2 members" implies any union type is ok if it has only two members. We only should give the idea that only nillable types are allowed. Also having index in the message is useful

Reply:
Will re-phrase the message. The index is automatically added by a higher level exception message so I did not add it here.